src/shadertools/src/lib/transpile-shader.js

```diff
@@ -0,0 +1,51 @@
+// Transpiles shader source code to target GLSL version
+// We always run transpiler even if same version
+export default function transpileShader(source, targetGLSLVersion, isVertex) {
+  switch (targetGLSLVersion) {
+  case 300:
+    return isVertex ? convertVertexShaderTo300(source) : convertFragmentShaderTo300(source);
+  case 100:
+    return isVertex ? convertVertexShaderTo100(source) : convertFragmentShaderTo100(source);
+  default:
+    throw new Error(`unknown GLSL version ${targetGLSLVersion}`);
+  }
+}
+
+function convertVertexShaderTo300(source) {
+  return source
+    .replace(/attribute\s+/g, 'in ')
+    .replace(/varying\s+/g, 'out ')
+    .replace(/texture2D\s+/g, 'texture')
+    .replace(/textureCube\s+/g, 'texture');
+
+    // Deal with fragColor
+    // .replace(/gl_fragColor/g, 'fragColor ');
+}
+
+function convertFragmentShaderTo300(source) {
+  return source
+    .replace(/varying\s+/g, 'in ')
+    .replace(/texture2D\(/g, 'texture(')
+    .replace(/textureCube\(/g, 'texture(');
+
+    // Deal with fragColor
+    // .replace(/gl_fragColor/g, 'fragColor ');
+}
+
+function convertVertexShaderTo100(source) {
+  // /gm - treats each line as a string, so that ^ matches after newlines
+  return source
+    .replace(/^in\s+/gm, 'attribute ')
+    .replace(/^out\s+/gm, 'varying ')
+    .replace(/texture\(/g, 'texture2D(');
+}
+
+function convertFragmentShaderTo100(source) {
+  // /gm - treats each line as a string, so that ^ matches after newlines
```

Comment:
Add a TODO for converting into `textureCube`?, this might be complicated as based on the sampler type (first argument) we have to replace `texture` with either `texture2D` or `textureCube`.

Reply:
Yes this is a problem. My thinking is that developers will need to use the `textureCube` call even in GLSL 3.00. since we currently always transpile (even 3.00 to 3.00), it will be transpiled to `texture` so it will work.

As you say, unless we do full parsing we won't know the type of the sampler.  I'll think about how to describe this in the docs. I think it is OK as cubemaps are not that common.